### PR TITLE
Adding in Initial extreme_tierra template

### PR DESCRIPTION
**Pull request by mayafox**
##### ISSUE TYPE
 - New Template Pull Request

##### COMPONENT
extreme_slx_show_process_cpu_all_partitions.textfsm, sh[[ow]] pr[[ocess]] c[[pu]] a[[ll-partitions]]
extreme_slx_show_process_memory_summary.textfsm, sh[[ow]] pr[[ocess]] m[[emory]] s[[ummary]]
extreme_slx_show_process_cpu_top.textfsm, sh[[ow]] pr[[ocess]] c[[pu]] t[[op]]
extreme_tierra_show_counters_interface_ethernet_all_qos_queues.textfsm, sh[[ow]] co[[unters]] int[[erface]] e[[thernet]] a[[ll]] q[[os]] q[[ueues]]
extreme_tierra_show_running-config_qos_forwarding-group_all.textfsm, sh[[ow]] ru[[nning-config]] q[[os]] f[[orwarding-group]] a[[ll]]
extreme_tierra_show_lldp_neighbors_interface_ethernet_all.textfsm, sh[[ow]] ll[[dp]] n[[eighbors]] i[[nterface]] e[[thernet]] all
extreme_tierra_show_running-config_interface_management_0.textfsm, sh[[ow]] ru[[nning-config]] int[[erface]] m[[anagement]] 0
extreme_tierra_show_running-config_interface_port-channel.textfsm, sh[[ow]] ru[[nning-config]] int[[erface]] p[[ort-channel]]
extreme_tierra_show_running-config_system_logging_service.textfsm, sh[[ow]] ru[[nning-config]] sy[[stem]] l[[ogging]] s[[ervice]]
extreme_tierra_show_counters_lldp_interface_ethernet_all.textfsm, sh[[ow]] co[[unters]] ll[[dp]] i[[nterface]] e[[thernet]] all
extreme_tierra_show_running-config_ipv6_access-list_all.textfsm, sh[[ow]] ru[[nning-config]] ipv6 access-list all
extreme_tierra_show_running-config_qos_forwarding-group.textfsm, sh[[ow]] ru[[nning-config]] qos forwarding-group
extreme_tierra_show_running-config_mac_access-list_all.textfsm, sh[[ow]] ru[[nning-config]] mac access-list all
extreme_tierra_show_running-config_system_logging_host.textfsm, sh[[ow]] ru[[nning-config]] system logging host
extreme_tierra_show_capture_packet_interface_ethernet.textfsm, sh[[ow]] ca[[pture]] p[[acket]] i[[nterface]] e[[thernet]]
extreme_tierra_show_counters_interface_ethernet_brief.textfsm, sh[[ow]] co[[unters]] int[[erface]] e[[thernet]] b[[rief]]
extreme_tierra_show_lldp_neighbors_interface_ethernet.textfsm, sh[[ow]] lldp neighbors interface ethernet
extreme_tierra_show_running-config_interface_ethernet.textfsm, sh[[ow]] ru[[nning-config]] interface ethernet
extreme_tierra_show_running-config_ip_access-list_all.textfsm, sh[[ow]] ru[[nning-config]] ip access-list a[[ll]]
extreme_tierra_show_running-config_tacacs-server_host.textfsm, sh[[ow]] ru[[nning-config]] tacacs-server h[[ost]]
extreme_tierra_show_counters_lldp_interface_ethernet.textfsm, sh[[ow]] co[[unters]] lldp interface ethernet
extreme_tierra_show_counters_interface_ethernet_all.textfsm, sh[[ow]] co[[unters]] int[[erface]] e[[thernet]] a[[ll]]
extreme_tierra_show_counters_interface_management_0.textfsm, sh[[ow]] co[[unters]] int[[erface]] m[[anagement]] 0
extreme_tierra_show_counters_interface_port-channel.textfsm, sh[[ow]] co[[unters]] int[[erface]] p[[ort-channel]]
extreme_tierra_show_running-config_ipv6_access-list.textfsm, sh[[ow]] ru[[nning-config]] ipv6 access-list
extreme_tierra_show_running-config_transport-tunnel.textfsm, sh[[ow]] ru[[nning-config]] t[[ransport-tunnel]]
extreme_tierra_show_running-config_listener-policy.textfsm, sh[[ow]] ru[[nning-config]] listener-policy
extreme_tierra_show_running-config_mac_access-list.textfsm, sh[[ow]] ru[[nning-config]] mac access-list
extreme_tierra_show_running-config_system_hostname.textfsm, sh[[ow]] ru[[nning-config]] system hostname
extreme_tierra_show_counters_link-fault-signaling.textfsm, sh[[ow]] co[[unters]] link-fault-signaling
extreme_tierra_show_counters_transport-tunnel_all.textfsm, sh[[ow]] co[[unters]] transport-tunnel all
extreme_tierra_show_grpc-server_gnmi_capabilities.textfsm, sh[[ow]] grpc-server gnmi capabilities
extreme_tierra_show_running-config_ip_access-list.textfsm, sh[[ow]] ru[[nning-config]] ip access-list
extreme_tierra_show_capture_packet_interface_all.textfsm, sh[[ow]] ca[[pture]] packet interface all
extreme_tierra_show_capture_packet_pcapfile-info.textfsm, sh[[ow]] ca[[pture]] packet pcapfile-info
extreme_tierra_show_interface_port-channel_brief.textfsm, sh[[ow]] interface port-channel brief
extreme_tierra_show_running-config_ingress-group.textfsm, sh[[ow]] ru[[nning-config]] ingress-group
extreme_tierra_show_counters_interface_ethernet.textfsm, sh[[ow]] co[[unters]] interface ethernet
extreme_tierra_show_grpc-server_gnmi_statistics.textfsm, sh[[ow]] grpc-server gnmi statistics
extreme_tierra_show_running-config_egress-group.textfsm, sh[[ow]] ru[[nning-config]] egress-group
extreme_tierra_show_counters_ingress-group_all.textfsm, sh[[ow]] co[[unters]] ingress-group all
extreme_tierra_show_running-config_access-list.textfsm, sh[[ow]] ru[[nning-config]] access-list
extreme_tierra_show_running-config_snmp-server.textfsm, sh[[ow]] ru[[nning-config]] snmp-server
extreme_tierra_show_counters_egress-group_all.textfsm, sh[[ow]] co[[unters]] egress-group all
extreme_tierra_show_counters_transport-tunnel.textfsm, sh[[ow]] co[[unters]] transport-tunnel
extreme_tierra_show_running-config_acl-config.textfsm, sh[[ow]] ru[[nning-config]] acl-config
extreme_tierra_show_running-config_ntp_server.textfsm, sh[[ow]] ru[[nning-config]] ntp server
extreme_tierra_show_qos_forwarding-group_all.textfsm, sh[[ow]] qos forwarding-group all
extreme_tierra_show_running-config_interface.textfsm, sh[[ow]] ru[[nning-config]] interface
extreme_tierra_show_running-config_route-map.textfsm, sh[[ow]] ru[[nning-config]] route-map
extreme_tierra_show_lacp_interface_ethernet.textfsm, sh[[ow]] lacp interface ethernet
extreme_tierra_show_running-config_username.textfsm, sh[[ow]] ru[[nning-config]] username
extreme_tierra_show_counters_ingress-group.textfsm, sh[[ow]] co[[unters]] ingress-group
extreme_tierra_show_crypto_ca_certificates.textfsm, sh[[ow]] crypto ca certificates
extreme_tierra_show_interface_management_0.textfsm, sh[[ow]] interface management 0
extreme_tierra_show_interface_port-channel.textfsm, sh[[ow]] interface port-channel
extreme_tierra_show_inventory_power-supply.textfsm, sh[[ow]] inventory power-supply
extreme_tierra_show_lacp_system-identifier.textfsm, sh[[ow]] lacp system-identifier
extreme_tierra_show_logging_audit_firmware.textfsm, sh[[ow]] logging audit firmware
extreme_tierra_show_logging_audit_security.textfsm, sh[[ow]] logging audit security
extreme_tierra_show_ntp_association_detail.textfsm, sh[[ow]] ntp association detail
extreme_tierra_show_sysinfo_sensor_voltage.textfsm, sh[[ow]] sysinfo sensor voltage
extreme_tierra_show_system_logging_service.textfsm, sh[[ow]] system logging service
extreme_tierra_show_capture_packet_config.textfsm, sh[[ow]] ca[[pture]] packet config
extreme_tierra_show_counters_egress-group.textfsm, sh[[ow]] co[[unters]] egress-group
extreme_tierra_show_counters_lldp_summary.textfsm, sh[[ow]] co[[unters]] lldp summary
extreme_tierra_show_running-config_banner.textfsm, sh[[ow]] ru[[nning-config]] banner
extreme_tierra_show_running-config_egress.textfsm, sh[[ow]] ru[[nning-config]] egress
extreme_tierra_show_running-config_ip_dns.textfsm, sh[[ow]] ru[[nning-config]] ip dns
extreme_tierra_show_running-config_mirror.textfsm, sh[[ow]] ru[[nning-config]] mirror
extreme_tierra_show_ipv6_access-list_all.textfsm, sh[[ow]] ipv6 access-list all
extreme_tierra_show_link-fault-signaling.textfsm, sh[[ow]] link-fault-signaling
extreme_tierra_show_logging_audit_config.textfsm, sh[[ow]] logging audit config
extreme_tierra_show_qos_forwarding-group.textfsm, sh[[ow]] qos forwarding-group
extreme_tierra_show_running-config_clock.textfsm, sh[[ow]] ru[[nning-config]] clock
extreme_tierra_show_running-config_encap.textfsm, sh[[ow]] ru[[nning-config]] encap
extreme_tierra_show_sysinfo_power-supply.textfsm, sh[[ow]] sysinfo power-supply
extreme_tierra_show_transport-tunnel_all.textfsm, sh[[ow]] transport-tunnel all
extreme_tierra_show_counters_egress_all.textfsm, sh[[ow]] co[[unters]] egress all
extreme_tierra_show_listener-policy_all.textfsm, sh[[ow]] listener-policy all
extreme_tierra_show_mac_access-list_all.textfsm, sh[[ow]] mac access-list all
extreme_tierra_show_system_logging_host.textfsm, sh[[ow]] system logging host
extreme_tierra_show_counters_encap_all.textfsm, sh[[ow]] co[[unters]] encap all
extreme_tierra_show_grpc-server_status.textfsm, sh[[ow]] grpc-server status
extreme_tierra_show_interface_ethernet.textfsm, sh[[ow]] interface ethernet
extreme_tierra_show_ip_access-list_all.textfsm, sh[[ow]] ip access-list all
extreme_tierra_show_running-config_aaa.textfsm, sh[[ow]] ru[[nning-config]] aaa
extreme_tierra_show_running-config_ntp.textfsm, sh[[ow]] ru[[nning-config]] ntp
extreme_tierra_show_running-config_qos.textfsm, sh[[ow]] ru[[nning-config]] qos
extreme_tierra_show_sysinfo_sensor_all.textfsm, sh[[ow]] sysinfo sensor all
extreme_tierra_show_sysinfo_sensor_cpu.textfsm, sh[[ow]] sysinfo sensor cpu
extreme_tierra_show_sysinfo_sensor_mem.textfsm, sh[[ow]] sysinfo sensor mem
extreme_tierra_show_ingress-group_all.textfsm, sh[[ow]] ingress-group all
extreme_tierra_show_sysinfo_sensor_lc.textfsm, sh[[ow]] sysinfo sensor lc
extreme_tierra_show_egress-group_all.textfsm, sh[[ow]] egress-group all
extreme_tierra_show_firmware_history.textfsm, sh[[ow]] firmware history
extreme_tierra_show_ipv6_access-list.textfsm, sh[[ow]] ipv6 access-list
extreme_tierra_show_transport-tunnel.textfsm, sh[[ow]] tr[[ansport-tunnel]]
extreme_tierra_show_counters_egress.textfsm, sh[[ow]] co[[unters]] egress
extreme_tierra_show_interface_brief.textfsm, sh[[ow]] interface brief
extreme_tierra_show_listener-policy.textfsm, sh[[ow]] listener-policy
extreme_tierra_show_mac_access-list.textfsm, sh[[ow]] mac access-list
extreme_tierra_show_media_interface.textfsm, sh[[ow]] me[[dia]] i[[nterface]]
extreme_tierra_show_media_supported.textfsm, sh[[ow]] me[[dia]] s[[upported]]
extreme_tierra_show_ntp_association.textfsm, sh[[ow]] ntp association
extreme_tierra_show_counters_encap.textfsm, sh[[ow]] co[[unters]] encap
extreme_tierra_show_inventory_slot.textfsm, sh[[ow]] inventory slot
extreme_tierra_show_ip_access-list.textfsm, sh[[ow]] ip access-list
extreme_tierra_show_lldp_neighbors.textfsm, sh[[ow]] lldp neighbors
extreme_tierra_show_media_detected.textfsm, sh[[ow]] me[[dia]] d[[etected]]
extreme_tierra_show_system_service.textfsm, sh[[ow]] system service
extreme_tierra_show_counters_lacp.textfsm, sh[[ow]] co[[unters]] lacp
extreme_tierra_show_ingress-group.textfsm, sh[[ow]] ingress-group
extreme_tierra_show_inventory_all.textfsm, sh[[ow]] inventory all
extreme_tierra_show_route-map_all.textfsm, sh[[ow]] route-map all
extreme_tierra_show_system_health.textfsm, sh[[ow]] system health
extreme_tierra_show_egress-group.textfsm, sh[[ow]] egress-group
extreme_tierra_show_logging_file.textfsm, sh[[ow]] logging file
extreme_tierra_show_sysinfo_slot.textfsm, sh[[ow]] sysinfo slot
extreme_tierra_show_tech-support.textfsm, sh[[ow]] tech-support
extreme_tierra_show_snmp-server.textfsm, sh[[ow]] snmp-server
extreme_tierra_show_sysinfo_fan.textfsm, sh[[ow]] sysinfo fan
extreme_tierra_show_sysinfo_led.textfsm, sh[[ow]] sysinfo led
extreme_tierra_show_acl-config.textfsm, sh[[ow]] acl-config
extreme_tierra_show_egress_all.textfsm, sh[[ow]] egress all
extreme_tierra_show_logging_id.textfsm, sh[[ow]] logging id
extreme_tierra_show_mirror_all.textfsm, sh[[ow]] mirror all
extreme_tierra_show_ntp_status.textfsm, sh[[ow]] ntp status
extreme_tierra_show_encap_all.textfsm, sh[[ow]] encap all
extreme_tierra_show_route-map.textfsm, sh[[ow]] route-map
extreme_tierra_show_firmware.textfsm, sh[[ow]] firmware
extreme_tierra_show_chassis.textfsm, sh[[ow]] ch[[assis]]
extreme_tierra_show_version.textfsm, sh[[ow]] v[[ersion]]
extreme_tierra_show_egress.textfsm, sh[[ow]] egress
extreme_tierra_show_ip_dns.textfsm, sh[[ow]] ip dns
extreme_tierra_show_mirror.textfsm, sh[[ow]] mirror
extreme_tierra_show_clock.textfsm, sh[[ow]] clock
extreme_tierra_show_encap.textfsm, sh[[ow]] encap
extreme_tierra_show_users.textfsm, sh[[ow]] use[[rs]]
extreme_tierra_show_lldp.textfsm, sh[[ow]] lldp
extreme_tierra_show_role.textfsm, sh[[ow]] role
extreme_tierra_show_qos.textfsm, sh[[ow]] qos
extreme_tierra_show_usb.textfsm, sh[[ow]] usb

##### SUMMARY
Template for Extreme Tierra OS commands, the command being show users, and a couple of extreme_slx templates 
Tox has run succesfully prior to checkin.

### CODE REVIEW

**Comment by mayafox** [comment]
_______________________________________________________________________________________________ summary _______________________________________________________________________________________________
SKIPPED:  py36: InterpreterNotFound: python3.6
SKIPPED:  py37: InterpreterNotFound: python3.7
  py38: commands succeeded
SKIPPED:  py39: InterpreterNotFound: python3.9
  black: commands succeeded
  flake8: commands succeeded
  yamllint: commands succeeded
  congratulations :)

File: tests/test_index_order.py [hunk 1]
```diff
@@ -58,7 +59,14 @@
...
 def check_order(
-    current_os, prior_os, cmd_len, prior_len, os_choices, used_os, cmd, prior_cmd
+    current_os,
+    prior_os,
+    cmd_len,
+    prior_len,
+    os_choices,
+    used_os,
+    cmd,
+    prior_cmd,
```
Comment (jvanderaa) [inline tests/test_index_order.py:69]: This is an unnecessary change.

File: tests/test_index_order.py [hunk 2]
```diff
@@ -121,7 +129,14 @@ def test_index_ordering():
...
         check_val, check_msg = check_order(
-            os, prior_os, cmd_len, prior_len, OS_CHOICES, used_os, cmd, prior_cmd
+            os,
+            prior_os,
+            cmd_len,
+            prior_len,
+            OS_CHOICES,
+            used_os,
+            cmd,
+            prior_cmd,
```
Comment (jvanderaa) [inline tests/test_index_order.py:139]: This is an unnecessary change, please revert.

File: tests/test_index_order.py [hunk 3]
```diff
@@ -32,7 +32,8 @@
     "dell_force10",
     "dell_powerconnect",
     "enterasys",
-    "extreme",
+    "extreme_slx",
+    "extreme_tierra",
```
Comment (jvanderaa) [inline tests/test_index_order.py:36]: This will be a breaking change. Can we get some more details on this? One other thought that I do not believe I am in favor of, but want to have the conversation. Should these match the Netmiko driver names?

**Review by jvanderaa — CHANGES_REQUESTED** [review]

File: tests/test_index_order.py (hunk 2 above)
Comment (mayafox) [reply tests/test_index_order.py:139]: Will do!

File: tests/test_index_order.py (hunk 1 above)
Comment (mayafox) [reply tests/test_index_order.py:69]: Will fix!

File: tests/test_index_order.py (hunk 3 above)
Comment (mayafox) [reply tests/test_index_order.py:36]: From the contributing section of the readme.md for the project.

```
Naming

The template should be named using: {{ vendor_os }}_{{ command_with_underscores }}.textfsm

Ex: cisco_ios_show_cdp_neighbors.textfsm
Note: The vendor name must be valid from the [os_choices](https://github.com/networktocode/ntc-templates/blob/master/tests/test_index_order.py#L59) tests, which is primarily based on [Netmiko](https://github.com/ktbyers/netmiko/tree/master/netmiko) list of supported vendors. New vendors added should adhere to vendor_os.

Ex: vmware_nsx
```

Extreme has recently acquired (2018) the rights to multiple product lines (MLX, SLXOS, NOS (VDX platforms), VYOS) and introduced the a new OS as well (TierraOS).

As 'extreme'  does not follow the stated format, and there are no 'extreme' templates present, this change brings ntc-templates inline with netmiko w/ respect to Extreme's product lines. and solves an issue that the test_index_order.py has due to not matching the vendor_os correctly when ran. 

Reference: [https://github.com/ktbyers/netmiko/blob/develop/netmiko/ssh_dispatcher.py] lines 173 to 182. Arguably, the `extreme` can be kept for backwards compatibility, but it causes issues here when it attempts to check the order of templates due to not following the stated format for contributing .

```
    "extreme": ExtremeExosSSH,
    "extreme_ers": ExtremeErsSSH,
    "extreme_exos": ExtremeExosSSH,
    "extreme_netiron": ExtremeNetironSSH,
    "extreme_nos": ExtremeNosSSH,
    "extreme_slx": ExtremeSlxSSH,
    "extreme_tierra": ExtremeTierraSSH,
    "extreme_vdx": ExtremeNosSSH,
    "extreme_vsp": ExtremeVspSSH,
    "extreme_wing": ExtremeWingSSH,
```

Comment (mayafox) [reply tests/test_index_order.py:36]: Ok, found a working solution, by putting the correctly formated vendor_os pairs ahead of the 'extreme' entry, the ordering test works correctly.  The other changes should be done as well.

**Comment by jmcgill298** [comment]
Replaced by #1691
